src/hooks/useClickOutside/index.ts

```diff
@@ -24,7 +24,7 @@ export default function useOnClickOutside({
   const documentRef = useRef(document.body);
 
   useEventListener({
-    eventName: "click",
+    eventName: "mousedown",
```

Comment:
@doronbrikman thanks for the PR, can you add this as a prop, like - `eventName` or `eventStage` or something that doesn't change the behaviour for all usecases

Reply:
@orrgottlieb - any reason why not to do this in all cases?

Reply:
I thought maybe we would like it to be the default value, do you want it to be `click` as default?
Also to which level we would like to pass this decision? Dialog level? or the usages of Dialog, like StatusPicker?, I would say Dialog

Reply:
My initial thought says that it should be mouseup for all, but if we are not sure about it, then I agree with Orr to provide this as an additional configuration and not change the original behavior for now

Reply:
`click` unlike `mouseup`/`mousedown` should happen on the same element. (it also should include the pointer events as well)

IMO - this indicates an intent to click outside of the element.

We have an issue in our monolith where a 3rd party is using the `mousedown` event, which doesn't indicate the intention of the user.

Because of that i thought of adding a configuration - `eventName` with the subset of `mousedown` `mouseup` `click`  - and add the relevant pointer events

Reply:
WDYT about that? @orrgottlieb 

> Also to which level we would like to pass this decision? Dialog level? or the usages of Dialog, like StatusPicker?, I would say Dialog

Reply:
@orrgottlieb - how should this be tested in the library to make sure there won't be regressions?

Reply:
As far as I understand, the proposed change still presents the same issue. We are modifying our component's behavior based on choices made by third parties, which may not necessarily be the 'correct' approach.
Why not just adding another closing trigger to the dialog onMouseUpOutside / onMouseDownOutside? @shlomitc @orrgottlieb

Reply:
Regarding tests: 
For the useOnClickOutside hook, I would suggest adding two tests that verify if the correct events are triggered for different event names.

As for the Dialog component, it depends on the chosen implementation approach. It might be easier to add interaction-based tests using Storybook to test the dialog's behavior (test that the dialog with "close by click outside ability" actually closing after click outside)

Alternatively, we can consider implementing regular Jest tests for the Dialog component, checking if both click and mouseup events cause the dialog disappear. However, if this proves to be challenging, utilizing Storybook for interaction testing might be simpler and good enough.

Reply:
Regarding the test: so sounds like the storybook interaction testing is better here. is there an example to get inspired by?

Regarding your suggestion - if the solution in this PR works without any regression - I think it will be better than exposing another configuration

Reply:
Tests: there are some example in our codebase in combobox component and useActiveDescendantListFocus
API: let's talk about it in a meeting?